src/ComposerGenerator.php

```diff
@@ -270,6 +280,12 @@ public function mergeFrameworkRequirements($rootComposer, $moduleComposer) {
 			$rootComposer['require']['silverstripe/cms'] = $coreConstraint;
 		}
 
+		// if we are testing a core module we need to force the composer to identify this as the core release we are
+		// targeting. This helps resolve dependencies on forked repos of our core modules
+		if ($this->isCoreModule($moduleComposer['name']) && isset($rootComposer['require'][$moduleComposer['name']])) {
+			$rootComposer['require'][$moduleComposer['name']] = $coreConstraint;
```

Comment:
Do you need to move / add this logic into `getModuleComposerConstraint` as well?